internal/pkg/config/config.go

```diff
@@ -188,46 +183,69 @@ type sourceMatcher struct {
 	Field   stringMatcher
 }
 
-// this type uses the default unmarshaller and mirrors configuration key-value pairs
-type rawSourceMatcher struct {
-	Package   *literalMatcher
-	Type      *literalMatcher
-	Field     *literalMatcher
-	PackageRE *regexp.Regexp
-	TypeRE    *regexp.Regexp
-	FieldRE   *regexp.Regexp
-}
-
 func (s *sourceMatcher) UnmarshalJSON(bytes []byte) error {
-	validSourceMatcherFields := []string{"package", "packageRE", "type", "typeRE", "field", "fieldRE"}
-	if err := validateFieldNames(&bytes, "sourceMatcher", validSourceMatcherFields); err != nil {
-		return err
-	}
+	// Unspecified matchers match vacuously
+	s.Package = vacuousMatcher{}
+	s.Type = vacuousMatcher{}
+	s.Field = vacuousMatcher{}
```

Comment:
Might read better as `= nil`, use ` != nil` checks below, and update `nil` to `vacuous` during final validation.  But it's a draft.